R/spliceHelperFunctions.R

```diff
@@ -224,9 +230,9 @@ annotateExonSkip <- function(spliceRng, fullRng, startRng, endRng){
 #' annotate exon splicing
 #' @noRd
 annotateExonSplice <- function(spliceRng, fullRng, startRng, endRng, strand){
-  splice.FullSplice.Rng <- expandMatchedRanges(spliceRng, fullRng)
-  start.Splice.Rng <- expandMatchedRanges(startRng, spliceRng)
-  end.Splice.Rng <- rep(unlist(endRng), elementNROWS(spliceRng))
+  splice.FullSplice.Rng <- expandSpliceRanges(spliceRng, fullRng)
+  start.Splice.Rng <- expandStartEndRanges(startRng, spliceRng)
+  end.Splice.Rng <- rep(endRng, elementNROWS(spliceRng))
   fullSpliceTable <- tibble(splice.FullSplice.start = start(splice.FullSplice.Rng),
```

Comment:
you can replace fullSpliceTable with 
```
  startMatch <- poverlaps(start(splice.FullSplice.Rng),mcols(splice.FullSplice.Rng)$matchRng) 
  endMatch <- poverlaps(end(splice.FullSplice.Rng),mcols(splice.FullSplice.Rng)$matchRng)
  exonStart <- endMatch & !startMatch
  exonEnd <-startMatch & !endMatch
```
this might not be faster but will be more readable. Can you think about a similar way to replace startEndSpliceTable?

Reply:
I have replaced startEndSpliceTable by calling findExon[Start/End]Extension functions.